tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_framework_version.py

```diff
@@ -21,135 +21,215 @@
 from tests.unit.sagemaker.cli.compatibility.v2.modifiers.ast_converter import ast_call
 
 
+class Template:
+    """Essentially a data class with a parametrized format method
+
+    Helper to interpolate various combinations of framework_version, py_version, and image
+    as expected by framework and model classes
+
+    TODO: use attrs package and eliminate the boilerplate...
+    """
+
+    def __init__(
+        self,
+        framework,
+        framework_version,
+        py_version,
+        framework_templates,
+        model_templates,
+        py_version_for_model=True,
+    ):
+        self.framework = framework
+        self.framework_version = framework_version
+        self.py_version = py_version
+        self.framework_templates = framework_templates
+        self.model_templates = model_templates
+        self.py_version_for_model = py_version_for_model
+
+    def constructors(self, versions=False, image=False):
+        return self._frameworks(versions, image) + self._models(versions, image)
+
+    def _frameworks(self, versions=False, image=False):
+        keywords = dict()
+        if image:
+            keywords["image_name"] = "my:image"
+        if versions:
+            keywords["framework_version"] = self.framework_version
+            keywords["py_version"] = self.py_version
+        return _format_templates(keywords, self.framework_templates)
+
+    def _models(self, versions=False, image=False):
+        keywords = dict()
+        if image:
+            keywords["image"] = "my:image"
+        if versions:
+            keywords["framework_version"] = self.framework_version
+            if self.py_version_for_model:
+                keywords["py_version"] = self.py_version
+        return _format_templates(keywords, self.model_templates)
+
+
+def _format_templates(keywords, templates):
+    args = ", ".join(
+        "{key}='{value}'".format(key=key, value=value) for key, value in keywords.items()
+    )
+    return [template.format(args) for template in templates]
+
+
+TEMPLATES = [
+    Template(
+        framework="TensorFlow",
+        framework_version="1.11.0",
+        py_version="py2",
+        framework_templates=[
+            "TensorFlow({})",
+            "sagemaker.tensorflow.TensorFlow({})",
+            "sagemaker.tensorflow.estimator.TensorFlow({})",
```

Comment:
(out of scope) have you thought about generating the variations as well? the only thing different from framework to framework for these is the framework name

Reply:
good call